test/python/transpiler/test_consolidate_blocks.py

```diff
@@ -428,6 +429,91 @@ def test_identity_1q_unitary_is_removed(self):
         pm = PassManager([Collect2qBlocks(), Collect1qRuns(), ConsolidateBlocks()])
         self.assertEqual(QuantumCircuit(5), pm.run(qc))
 
+    def test_descent_into_control_flow(self):
+        """Test consolidation in blocks when control flow op is the same as at top level."""
+        qc = QuantumCircuit(2, 1)
+        u2gate1 = U2Gate(-1.2, np.pi)
+        u2gate2 = U2Gate(-3.4, np.pi)
+        qc.append(u2gate1, [0])
+        qc.append(u2gate2, [1])
+        qc.cx(0, 1)
+        qc.cx(1, 0)
+
+        pass_manager = PassManager()
+        pass_manager.append(Collect2qBlocks())
+        pass_manager.append(ConsolidateBlocks(force_consolidate=True))
+        result_top = pass_manager.run(qc)
+
+        qc_control_flow = QuantumCircuit(2, 1)
+        qc_block = copy.deepcopy(qc)
+
+        qc_block = QuantumCircuit(qc.qubits, qc.clbits)
```

Comment:
Why clone `qc` if it's about to be overwritten?  It also shouldn't be necessary for `qc_block` to use the same bit instances as `qc`.

Fwiw, cloning a `QuantumCircuit` can / should be done with `QuantumCircuit.copy`.

Reply:
Looks like cruft. Coding while tired (!)

Reply:
Done in 248479d (and the following commit)